core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -603,12 +623,34 @@ default void finishStatisticsCollection(ConnectorSession session, ConnectorTable
      *     new TrinoException(NOT_SUPPORTED, "This connector does not support query retries")
      * </pre>
      * unless {@code retryMode} is set to {@code NO_RETRIES}.
+     *
+     * @deprecated use {@link #beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional layout, RetryMode retryMode, boolean replace)}
      */
+    @Deprecated
     default ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorTableLayout> layout, RetryMode retryMode)
     {
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating tables with data");
     }
 
+    /**
+     * Begin the atomic creation of a table with data.
+     *
+     * <p/>
+     * If connector does not support execution with retries, the method should throw:
+     * <pre>
+     *     new TrinoException(NOT_SUPPORTED, "This connector does not support query retries")
+     * </pre>
+     * unless {@code retryMode} is set to {@code NO_RETRIES}.
+     */
+    default ConnectorOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, Optional<ConnectorTableLayout> layout, RetryMode retryMode, boolean replace)
```

Comment:
Shouldn't we maintain the `SaveMode` here ?

Reply:
Based on the Iceberg docs for Spark integration - https://iceberg.apache.org/docs/latest/spark-ddl/#replace-table--as-select
> The new table properties in the REPLACE TABLE command will be merged with any existing table properties. The existing table properties will be updated if changed else they are preserved.

Does the connector implementation handles the property updation ? Should we have additional test coverage for Iceberg ?

Reply:
I think we shouldn't do that because we don't have the `INSERT OVERWRITE` syntax. Also this will make the feature a lot less usable in dbt (if the user completely changes the implementation or removes partitioning)

If the `CREATE OR REPLACE` syntax is used, the user should provide the relevant table properties, we don't assume anything. I think it is more straight forward to use in this way.

Reply:
But even on computing brand new property, can we replace this boolean with `SaveMode` enum.

Reply:
If we need to be able to distinguish between "fail if it already exists" vs "ignore if it already exists" modes, yes, then this needs use `SaveMode`.

Reply:
We don't need this distinction. That is handled in `default void createTable(ConnectorSession session, ConnectorTableMetadata tableMetadata, SaveMode saveMode)`. That is also the reason why i used the boolean instead of the enum here.

Reply:
Even if after we check if a table exists or not when we invoke `CreateTableTask` - why do we pass the `ignoreExisting` flag ? Do we need this flag in the future (once we replace it with enum)